core/src/scene/styleParam.h

```diff
@@ -60,6 +60,28 @@ constexpr size_t StyleParamKeySize = static_cast<size_t>(StyleParamKey::NUM_ELEM
 
 enum class Unit { pixel, milliseconds, meter, seconds };
 
+static std::string unitString(Unit unit) {
+    switch(unit) {
+        case Unit::pixel: return "pixel";
+        case Unit::milliseconds: return "milliseconds";
+        case Unit::meter: return "meter";
+        case Unit::seconds: return "seconds";
+        default: return "undefined";
+    }
+}
+
+template <typename T, int size>
+struct UnitVec {
+    T value = T(0.0);
+    Unit units[size];
```

Comment:
*facepalm* my bad, it makes no sense to index into a type name haha. `sizeof(value)/sizeof(value[0])` should work though. Then we can remove the `size` template parameter.